### ISSUE TITLE
coturn log file pile up due to "TCP socket closed remotely" messages in log file

### ISSUE THREAD

**Issue by Govind10g** [coturn/coturn#1130]
Hi Team, 

We found that "TCP socket closed remotely" logs are continuously generating in coturn.log file which is creating a huge log file and it is difficult to debug the issue because of that. 

Below is the same of TCP socket closed remotely logs in coturn.log file :-

```
23: : session 001000000000000030: TCP socket closed remotely 10.41.143.22:57711
23: : session 001000000000000030: usage: realm=<xyz.com>, username=<>, rp=0, rb=0, sp=0, sb=0
23: : session 001000000000000030: peer usage: realm=<xyz.com>, username=<>, rp=0, rb=0, sp=0, sb=0
23: : session 001000000000000030: closed (2nd stage), user <> realm <xyz.com> origin <>, local 10.41.15.88:443, remote10.41.14.22:57711, reason: TCP connection closed by client (callback)
24: : IPv4. tcp or tls connected to: 10.41.149.56:34252
24: : session 000000000000000027: TCP socket closed remotely 10.41.147.56:34252
24: : session 000000000000000027: usage: realm=<xyz.com>, username=<>, rp=0, rb=0, sp=0, sb=0
24: : session 000000000000000027: peer usage: realm=<xyz.com>, username=<>, rp=0, rb=0, sp=0, sb=0
24: : session 000000000000000027: closed (2nd stage), user <> realm <xyz.com> origin <>, local 10.41.15.88:443, remote10.41.148.56:34252, reason: TCP connection closed by client (callback)
24: : IPv4. tcp or tls connected to: 10.41.145.22:18943
2 
```

**How we can get rid from the above log from coturn.log file ?**
@eakraly

**Comment by Govind10g** [comment]
To add more on this, we are using "verbose" flag in the config file and with the help of that we are getting call logs as well as above mentioned logs.

But when we disabled "verbose" no logs are coming in the log file. :(

**Comment by eakraly** [comment]
Hi @Govind10g 

It does not look there is anything wrong with the flow in the log provided (and this is not what you asked but just making sure)
Those logs are printed when `--verbose` or `--Verbose` are on - you may want to try and run without those flags.

**Comment by Govind10g** [comment]
Hi @eakraly ,

Yes there is nothing wrong with the flow, but that above message logs are continuously writing in the coturn log file and bcoz of that it is creating a huge file , I mean the size of coturn file.

**Comment by eakraly** [comment]
Not really. We have a longer term plan to revamp logging but nothing immediately I could suggest (unless you are willing to patch code and build a custom version)

**Comment by ggarber** [comment]
@Govind10g Adding to what Pavel said you should be able to filter those logs for debugging purposes with `grep -v "TCP socket closed" coturn.log`

Also if you use some type of centralized logging you should be able to apply that filtering before forwarding the logs.

Hope that solves your immediate needs.

**Comment by wolmi** [comment]
We are also getting that logs because of healthchecks running in front of the coturn instances.

Our scenario was initially configured to use TCP health checks to the TLS port to allow load balancers to distribute the traffic in two stages: first in DNS, and second at the network level.

Also, we are using a centralized log correlation SaaS, which has an extra cost to the number of logs ingested.

The solution was to use coturn with prometheus enabled, we have compiled our own to be used in Debian instances, and changed the healthchecks to try port 9641 of prometheus in the `/` that was added to have a more reliable endpoint in HTTP for healthchecks.
